### PR TITLE
Axissync merge

### PR DESCRIPTION
**Pull request by nightwolfdu**
Hello again! 
I worked on the issue of multi axis in Victory Chart recently and got big fail in domain/scale separating and delivering into all components. 
But with it, i wrote some code for axises synchronization. (identical number of ticks and labels in each axis, with normal tickInterval)
I moved it in special helper and suggest you use it separately or in Victory Chart, if you would support MultiAxis in the near future.
axis-sync

Alignment algorithm:
1) getTicks 
4
3
2 2
1 1 1
0 0 0
2) get max tick count :
max(..ticks)
5
3) get tick interval and add need count of ticks as maxCount - curCount
4 4 4
3 3 3 
2 2 1
1 1 1
0 0 0

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,6 +30,7 @@
     "builder-victory-component": "^2.4.0",
     "d3-scale": "^1.0.0",
     "d3-voronoi": "^1.0.0",
+    "d3": "^4.2.2",
```
Comment (boygirl) [inline package.json]: It looks like you're only using d3-scale, so no need to include all of d3

File: src/helpers/axissync.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
+import { TextSize } from "victory-core";
+import Axis from "./axis";
+import VictoryAxis from "../components/victory-axis/victory-axis.js";
+import AxisHelper from "../components/victory-axis/helper-methods.js";
+import { defaults, range, groupBy, flatten, merge } from "lodash";
+import * as d3 from "d3";
```
Comment (boygirl) [inline src/helpers/axissync.js]: use d3-scale

**Comment by boygirl** [comment]
@nightwolfdu I like this enhancement as a step towards supporting multiple axes in VictoryChart, but it seems like it does two things. 1) figures out ideal ticks based on label size, and 2) syncs two (or more?) axes. I think there's a place for each of these, but I'd like to see them split apart. I need to think a little bit more about when each of these things should be applied

**Comment by nightwolfdu** [comment]
TickStep was defined in d3-array, and i can't find it in d3-scale.
Rewrite to d3-array.
I split code on 2 public function and did some refactoring, but i not like their api.
This helper gives some profit in itself, but I do not insist on merge it in _release_ branch.

**Comment by boygirl** [comment]
@nightwolfdu how would you feel about a `smart-axis` branch to keep these experiments in until we settle on what should be included in VictoryChart?

**Comment by nightwolfdu** [comment]
@boygirl Lauren, as you wish.
Now I'm busy at work and do not have time to finish this branch (

**Comment by boygirl** [comment]
That's okay. I'm going merge this into the branch I made. I'll let you know if I start working on multi-axis support.

**Comment by nightwolfdu** [comment]
Thanks a lot.

**Pull request merged**